libseccomp-sys/src/lib.rs

```diff
@@ -3,22 +3,194 @@
 // Copyright 2021 Sony Group Corporation
 //
 
-#![allow(dead_code)]
-#![allow(non_upper_case_globals)]
 #![allow(non_camel_case_types)]
 #![allow(non_snake_case)]
-#![allow(clippy::all)]
 
-/// Manually support SCMP_ACT_ERRNO and SCMP_ACT_TRACE macro
-/// because the bindgen cannot expand the macros correctly.
-///
-/// Return the specified error code
-pub fn SCMP_ACT_ERRNO(x: u16) -> u32 {
-    0x00050000_u32 | u32::from(x)
+use std::os::raw::*;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct scmp_version {
+    pub major: c_uint,
+    pub minor: c_uint,
+    pub micro: c_uint,
+}
+
+pub type scmp_filter_ctx = *mut c_void;
+pub type const_scmp_filter_ctx = *const c_void;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub enum scmp_filter_attr {
+    _SCMP_FLTATR_MIN = 0,
+    SCMP_FLTATR_ACT_DEFAULT = 1,
+    SCMP_FLTATR_ACT_BADARCH = 2,
+    SCMP_FLTATR_CTL_NNP = 3,
+    SCMP_FLTATR_CTL_TSYNC = 4,
+    SCMP_FLTATR_API_TSKIP = 5,
+    SCMP_FLTATR_CTL_LOG = 6,
+    SCMP_FLTATR_CTL_SSB = 7,
+    SCMP_FLTATR_CTL_OPTIMIZE = 8,
+    SCMP_FLTATR_API_SYSRAWRC = 9,
+    _SCMP_FLTATR_MAX,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub enum scmp_compare {
+    _SCMP_CMP_MIN = 0,
+    SCMP_CMP_NE = 1,
+    SCMP_CMP_LT = 2,
+    SCMP_CMP_LE = 3,
+    SCMP_CMP_EQ = 4,
+    SCMP_CMP_GE = 5,
+    SCMP_CMP_GT = 6,
+    SCMP_CMP_MASKED_EQ = 7,
+    _SCMP_CMP_MAX,
+}
+
+pub type scmp_datum_t = u64;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct scmp_arg_cmp {
+    pub arg: c_uint,
+    pub op: scmp_compare,
+    pub datum_a: scmp_datum_t,
+    pub datum_b: scmp_datum_t,
+}
+
```

Comment:
Could you add `pub const SECCOMP_USER_NOTIF_FLAG_CONTINUE: u32 = 1;`?
I noticed that we need the flag for Seccomp Notification.

Reply:
It would be better to generate `libseccomp-sys` with `-allowlist-var=".*SECCOMP_.*"`.

Reply:
Do we need any other `linux/seccomp.h` constants?

Reply:
I think I just add all the constants from `linux/seccomp.h` as well (except for `SECCOMP_IO*`).

Reply:
Yes, could you try to add all the constants from` linux/seccomp.h` (except for SECCOMP_IO*)?